clarity_ext/service/step_logger_service.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import time
+from clarity_ext.service.file_service import SharedFileNotFound
+
+
+class StepLoggerService:
+    """
+    The logger service provides logging for files attached in the step.
+
+    Extension writers are exposed to this class via the `ExtensionContext`, which supplies direct
+    access to both logging artifacts, without them having to do anything but ensuring that
+    a file is defined on the step
+    """
+    def __init__(self, step_logger_name, file_service, raise_if_not_found=False, append=True):
+        self.core_logger = logging.getLogger(__name__)
+        self.step_logger_name = step_logger_name
+        self.file_service = file_service
+
+        # Use Windows line endings for now, since most clients are currently Windows.
+        # TODO: This should be configurable.
+        self.NEW_LINE = "\r\n"
+        try:
```

Comment:
Yes, the idea with that method was that it specifies what type of newlines there should be used in driverfiles, specified by extension writers. I don't see any clear reason for reusing it though, since this is to be configured by administrators.